src/containers/TaskRuns/TaskRuns.js

```diff
@@ -0,0 +1,213 @@
+/*
+Copyright 2019 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { InlineNotification } from 'carbon-components-react';
+
+import { getTasks, getTaskRuns } from '../../api';
+
+import RunHeader from '../../components/RunHeader';
+import StepDetails from '../../components/StepDetails';
+import TaskTree from '../../components/TaskTree';
+import { getStatus, taskRunStep, selectedTaskRun } from '../../utils';
+
+import '../../components/Run/Run.scss';
+
+/* istanbul ignore next */
+class TaskRunsContainer extends Component {
+  state = {
+    notification: null,
+    loading: true,
+    selectedStepId: null,
+    selectedTaskId: null,
+    taskRuns: []
+  };
+
+  async componentDidMount() {
+    try {
+      const { match } = this.props;
+      const { taskName } = match.params;
+      await this.loadTaskRuns(taskName);
+    } catch (error) {
+      let message = error;
+      if (error.response) {
+        message = error.response.status === 404 ? 'Not Found' : 'Error';
+      }
+      this.setState({
+        notification: { kind: 'error', message },
+        loading: false
+      });
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    const { match } = this.props;
+    const { taskName } = match.params;
+    if (taskName !== prevProps.match.params.taskName) {
+      this.loadTaskRuns(taskName);
+    }
+  }
+
+  handleTaskSelected = (selectedTaskId, selectedStepId) => {
+    this.setState({ selectedStepId, selectedTaskId });
+  };
+
+  async loadTaskRuns(selectedTaskName) {
+    let notification;
+    const tasks = await getTasks();
+    const task = tasks.find(
+      currentTask => currentTask.metadata.name === selectedTaskName
+    );
+    let taskRuns = await getTaskRuns();
+    taskRuns = taskRuns
+      .filter(
+        taskRun =>
+          taskRun.spec.taskRef && taskRun.spec.taskRef.name === selectedTaskName
```

Comment:
TODO: discuss additional back end API to retrieve TaskRuns for given Task name.
Not required for this initial PR but should follow up separately.